hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/artifacts/smtable/SoftwareModuleAddUpdateWindow.java

```diff
@@ -360,4 +332,13 @@ private boolean mandatoryCheck(final String name, final String version, final St
         }
         return isValid;
     }
+
+    private void save() {
+        if (editSwModule) {
+            updateSwModule();
+        } else {
+            /* add new or update software module */
```

Comment:
The comment is misleading. You update the software module in line 338, aren't you?